tests/src/OneLogin/saml2_tests/idp_metadata_parser_test.py

```diff
@@ -51,7 +51,7 @@ def testParseRemote(self):
 
         data = OneLogin_Saml2_IdPMetadataParser.parse_remote('https://www.testshib.org/metadata/testshib-providers.xml')
         self.assertTrue(data is not None and data is not {})
-        expected_data = {'sp': {'NameIDFormat': 'urn:mace:shibboleth:1.0:nameIdentifier'}, 'idp': {'singleSignOnService': {'url': 'https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO'}, 'entityId': 'https://idp.testshib.org/idp/shibboleth'}}
+        expected_data = {'sp': {'NameIDFormat': 'urn:mace:shibboleth:1.0:nameIdentifier'}, 'idp': {'singleSignOnService': {'url': 'https://idp.testshib.org/idp/profile/SAML2/Redirect/SSO', 'binding': 'urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect'}, 'entityId': 'https://idp.testshib.org/idp/shibboleth'}}
         self.assertEqual(expected_data, data)
 
     def testParse(self):
```

Comment:
I absolutely agree with you! We should work on this. "Cover all scenarios" is a huge task in this case :-), because there is an infinite amount of example XML documents we could use.

The current method of pulling `testshib-providers.xml` can be extended by pulling in more example XML documents -- do you have specific proposals? That is, is there something specific you want me to add to this PR? Maybe I should transform `expected_data` to be human-readable JSON, too (as in the merge test case), so that we can at least work with useful diffs on the reference data in future PRs.